File: easy.server.demo/easy.server.boot.demo/src/main/java/io/github/jinlongliao/easy/server/boot/demo/logic/annotation/Logic.java
package io.github.jinlongliao.easy.server.boot.demo.logic.annotation;

import io.github.jinlongliao.easy.server.mapper.annotation.GeneratorMethod;

import java.lang.annotation.*;

/**
 * 自定义
 *
 * @author: liaojinlong
 * @date: 2023/1/31 22:19
 */
@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@GeneratorMethod
@Documented
public @interface Logic {
    /**
     * 标记
     *
     * @return /
     */
    MsgId[] value();
}

File: easy.server.demo/easy.server.kotlin.demo/src/main/kotlin/io/github/jinlongliao/easy/server/kotlin/demo/logic/annotation/Logic.kt
package io.github.jinlongliao.easy.server.kotlin.demo.logic.annotation

import io.github.jinlongliao.easy.server.mapper.annotation.GeneratorMethod

/**
 * 自定义
 *
 * @author: liaojinlong
 * @date: 2023/1/31 22:19
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(
    AnnotationRetention.RUNTIME
)
@GeneratorMethod
@MustBeDocumented
annotation class Logic(
    /**
     * 标记
     *
     * @return /
     */
    vararg val value: MsgId
)
